win32ss/gdi/ntgdi/freetype.c

```diff
@@ -1577,16 +1599,36 @@ IntGdiAddFontResourceEx(PUNICODE_STRING FileName, DWORD Characteristics,
     PVOID Buffer = NULL;
     IO_STATUS_BLOCK Iosb;
     PVOID SectionObject;
-    SIZE_T ViewSize = 0;
+    SIZE_T ViewSize = 0, Length;
     LARGE_INTEGER SectionSize;
     OBJECT_ATTRIBUTES ObjectAttributes;
     GDI_LOAD_FONT LoadFont;
     INT FontCount;
     HANDLE KeyHandle;
+    UNICODE_STRING PathName;
+    WCHAR szPath[MAX_PATH];
     static const UNICODE_STRING TrueTypePostfix = RTL_CONSTANT_STRING(L" (TrueType)");
 
+    /* Build PathName */
+    Length = wcslen(SharedUserData->NtSystemRoot);
+    RtlStringCbCopyNW(szPath, sizeof(szPath), FileName->Buffer,
```

Comment:
This seems superfluous. Why not compare against `FileName->Buffer` directly instead of copying it into `szPath`?

Reply:
OK.